### ISSUE TITLE
Encourage use of `importlib.metadata.version`

### ISSUE THREAD

**Issue by jakkdl** [astral-sh/ruff#15671]
### Description

I only recently found out about [importlib.metadata.version](https://docs.python.org/3/library/importlib.metadata.html#importlib.metadata.version) as a way of getting the installed version of a package; as opposed to reading some global `__version__` variable that may or [may not be](https://github.com/python-trio/trio/pull/3186) properly exported.
I would love a rule that encouraged the use of the more portable & reliable importlib variant, although I have a feeling that a simple rule that detected any `from ... import __version__` would raise a bunch of false alarms on packages that export it as ways of versioning ... other things than the package itself. (maybe only trigger if importing from the top namespace of a module?) Or idk other weird stuff.

Anyway, thought I'd throw out the idea anyway and see if it sticks.

**Comment by AlexWaygood** [comment]
This might be controversial for authors of CLIs, since `importlib.metadata` has historically had quite a high import time. Importing `importlib.metadata` and using `importlib.metadata.version()` could mean that running `<your_cli_app> --version` takes longer if it uses `importlib.metadata` than it would if it just checked `__version__`

**Comment by jakkdl** [comment]
> This might be controversial for authors of CLIs, since `importlib.metadata` has historically had quite a high import time. Importing `importlib.metadata` and using `importlib.metadata.version()` could mean that running `<your_cli_app> --version` takes longer if it uses `importlib.metadata` than it would if it just checked `__version__`

I was thinking of it more when interfacing with 3rd-party packages rather than within-package use; so you you'd probably want to exclude those imports.

But it definitely is slower, I did some quick benchmarks with
```python
import six
import importlib.metadata
print(importlib.metadata.version('six')
```
vs
```python
import six
print(six.__version__)
```

and the former averaged .0642s and the latter .0174 for a ~4x difference (py3.13; on py3.9 I got .049 and .022)

**Comment by flying-sheep** [comment]
So speed is basically just a concern for CLIs getting their own version. Also on Python 3.14 for me, the difference is ~30ms which lies well below lag perception level for a CLI.

I think it’s fair to say that when you want to get any *other* packages’ version, using a standard instead of a convention is a good idea.

And we already know which packages are first-party (configurable via [`lint.isort.known-first-party`](https://docs.astral.sh/ruff/settings/#lint_isort_known-first-party), so we can just suggest it for packages other than ourselves.
